src/components/PopupWithForm.js

```diff
@@ -4,27 +4,33 @@ class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
     super({ popupSelector });
     this._popupForm = this._modalEl.querySelector(".modal__form");
+    this._submitButton = this._modalEl.querySelector(".modal__save-button");
     this._handleFormSubmit = handleFormSubmit;
     this._list = [...this._popupForm.querySelectorAll(".modal__input")];
   }
 
+  open() {
+    this._submitButton.textContent = "Save";
+    super.open();
+  }
+
   close() {
     this._popupForm.reset();
     super.close();
   }
 
-  // setInputValues(data) {
-  //   this._inputList.forEach((input) => {
-  //     input.value = data[input.name];
-  //   });
-  // }
+  setLoading(loading) {
+    if (loading) {
+      this._submitButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```

Reply:
This made it much cleaner thank you.